lsdlabs/MovieJournalMVC
# Movie review addition

## What does this PR do?

Card: https://trello.com/c/3WBJlSNA

This PR allows a user to click the "+" button and add a dummy film, replacing the default NSDate type.  At this point a user cannot type in a film or a review.
 
## How was it done?

I started creating a `MovieReview` class to hold a movie review.  I changed the default `objects` array to use shared storage (`Store` class) and fixed the resulting compiler errors because `objects` is no longer of type `NSDate`.  Then, I removed the `NSDate` typecast in prepare-for-segue and table view `cellForRowAt` functions.

At this point, the compiler complained about `controller.detailItem = object` because I couldn't assign a `MovieReview` to type `NSDate.` . To start resolving this, I created and connected movie title and review labels, changed the `detailItem` variable from `NSDate` to an optional `MovieReview`, and altered the `DetailViewController's` `configureView()` function’s conditional unwrapping to use the custom UILabels.

Then the compiler complained that the cells weren't set up properly at `(cell.textLabel!.text = object.description)` giving me a warning of `Value of type 'MovieReview' has no member 'description'`.  To resolve this issue, I create a Swift class for defining how cells are defined called `MovieReviewCell` and connected this class to the prototype cell.

I returned to the master view controller to set up the generation of cells and typecasted the cell as `MovieReviewCell`, and in `cellForRowAt` function I replaced the default `cell.textLabel!.text = object.description` with `cell.movieTitleLabel.text! = object.title`. 

Not ideal, but I set the cell properties set to `MovieReview` object properties.  Last but not least, I incremented the counter because I forgot to do that to add new fake movie values.

## How can one test it?
Run the app and check to see if you click the `+` button that a new movie is added.  Once you click on a movie, you should be brought to a detail view with the fake movie information.

## Screenshot(s) (Optional)
![image](https://user-images.githubusercontent.com/42355058/54287839-d008a700-457c-11e9-999f-edeac8bbd4f3.png)

![image](https://user-images.githubusercontent.com/42355058/54287870-dd259600-457c-11e9-84e6-fa36d37ea9aa.png)
